sqlglot/expressions.py

```diff
@@ -1006,6 +1006,31 @@ class Create(Expression):
         "clone": False,
     }
 
+    @property
+    def ctes(self):
```

Comment:
should we move these helper methods to a shared base class and have this applied to Insert and Update or other DDLs? 

then in the optimizer you can check for the base class instead of just Create